src/plugins/intel_gpu/src/kernel_selector/cl_kernels/gather_ref.cl

```diff
@@ -54,7 +54,11 @@ KERNEL(gather_ref)(
     const uint dictionary_idx = GET_DICTIONARY_INDEX(DICTIONARY_INDEX_ORDER);
     const uint output_idx = GET_INDEX(OUTPUT,,ORDER);
 
+#if AXIS_DIM
+    INPUT0_TYPE val = (INPUT_AXIS_INDEX >= 0 && INPUT_AXIS_INDEX < AXIS_DIM) ? dictionary[dictionary_idx] : 0;
```

Comment:
shouldn't we support this case in shape agnostic kernel too? 
i.e., shouldnt we check the actual size in shape_info and compare?